File: 백준/Silver/11478. 서로 다른 부분 문자열의 개수/서로 다른 부분 문자열의 개수.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.HashSet;
import java.util.Set;

public class Main {
    public static void main(String[] args) throws Exception {
       BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
       BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        String s = br.readLine();
        Set<String> set = new HashSet<>();

        for  (int i = 0; i < s.length(); i++) {
            String ans = "";
            for (int j = i; j < s.length(); j++) {
                ans += s.charAt(j);
                set.add(ans);
            }
        }
        StringBuilder sb = new StringBuilder();
        sb.append(set.size());

        bw.write(sb.toString());
        bw.flush();
        bw.close();
    }
}

File: 백준/Silver/11478. 서로 다른 부분 문자열의 개수/서로 다른 부분 문자열의 개수.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val bw = BufferedWriter(OutputStreamWriter(System.out))

    val s = br.readLine()
    val set = mutableSetOf<String>()

    for (i in 0 until s.length) {
        var ans = ""
        for (j in i until s.length) {
            ans += s[j]
            set.add(ans)
        }
    }

    val sb = StringBuilder()
    sb.append(set.size)

    bw.write(sb.toString())
    bw.flush()
    bw.close()
}